File: PHP/CountOperationsToObtainZeroTest.php
<?php
/**
 * @version PHP 8.2.22
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit CountOperationsToObtainZeroTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("CountOperationsToObtainZero.php");

class CountOperationsToObtainZeroTest extends TestCase {

    public function testCountOperationsToObtainZero() {
        $obj = new CountOperationsToObtainZero();
        //Test 1
        $want = 3;
        $got = $obj->countOperations( 2, 3 );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 1;
        $got = $obj->countOperations( 10, 10 );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/CountOperationsToObtainZero_test.go
/**
 * @version GO 1.23.5
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestCountOperationsToObtainZero
 */
package GO

import (
	"reflect"
	"testing"
)

func TestCountOperationsToObtainZero(t *testing.T) {
	tables := []struct {
		want int
		num1 int
		num2 int
	}{
		{3, 2, 3},
		{1, 10, 10},
	}

	for _, table := range tables {
		got := countOperations(table.num1, table.num2)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
